File: examples/python-example/greet.py
"""Minimal module with exactly one branch to cover.

There is deliberately no ``if __name__ == "__main__"`` block, so every line is
reachable from the tests and coverage lands at 100%.
"""


def greet(name: str = "") -> str:
    """Return a greeting for ``name``; an empty name greets the world."""
    if not name:
        return "Hello, world!"
    return f"Hello, {name}!"

File: examples/go-example/greet.go
// Package greet is the smallest possible library that still has a branch to
// cover. It is deliberately a library (no main function) so `go build ./...`
// succeeds and statement coverage can reach 100%.
package greet

import "fmt"

// Greet returns a greeting for name. An empty name greets the world.
func Greet(name string) string {
	if name == "" {
		return "Hello, world!"
	}
	return fmt.Sprintf("Hello, %s!", name)
}
